internal/v1/server_test.go

```diff
@@ -191,7 +191,10 @@ func startServer(t *testing.T, tscc *testServerClientsConf, conf *ServerConfig)
 	require.NoError(t, err)
 	// execute in parallel b/c .Run() will block execution
 	go func() {
-		_ = echoServer.Start("localhost:8086")
+		err = echoServer.Start("localhost:8086")
+		if err.Error() != "http: Server closed" {
```

Comment:
ServerClose always happens on regular shutdown...
I already did implement with `GetFreePort()` (locally) but stopped after changing ~1000 strings to take a dynamic host&port 😑
Will do this "major rework" at some other point in time

Reply:
I did a quick sed experiement and opened https://github.com/osbuild/image-builder/pull/1303 which may help, I did not really check much, it was mostly sed doing it's thing but it might be useful. No getFreePort there but that should be easy to hook up once the other pieces are there.

It also include the tiny `http.ErrServerClosed` tweak - hope this is useful.